src/main/java/de/tum/in/www1/artemis/service/connectors/BambooService.java

```diff
@@ -452,7 +452,20 @@ public Result onBuildCompletedNew(ProgrammingExerciseParticipation participation
             final var hasArtifact = buildResult.getBuild().isArtifact();
             programmingSubmission.setBuildArtifact(hasArtifact);
             programmingSubmission.setBuildFailed(result.getResultString().equals("No tests found"));
-            programmingSubmissionRepository.save(programmingSubmission);
+
+            programmingSubmission = programmingSubmissionRepository.save(programmingSubmission);
+
+            Optional<ProgrammingSubmission> optionalProgrammingSubmission = programmingSubmissionRepository.findWithEagerBuildLogEntries(programmingSubmission.getId());
+            if (optionalProgrammingSubmission.isPresent()) {
+                programmingSubmission = optionalProgrammingSubmission.get();
+                // Store logs into database. Append logs of multiple jobs.
```

Comment:
I am not sure if this is a good approach. In the future, we will have only one programming submission per commit (that was pushed to the remote). Then it could happen that multiple builds add multiple build logs.
Users might see the same build errors multiple times then.

I would suggest that we simply replace all build logs here. In case they are identical, this makes more sense.
In case they have changed, the old ones might not be valid any more.

Reply:
I see your point, however this comment is regarding multiple jobs within one build (-> one submission).
We currently only use one job but as the bamboo plugin offers a set of jobs, we should take log entries of every job into account.

So I suggest to clear the log entries before the for-loop and then adding the log entries of all jobs (although we most likely only have one) to the cleared list.

Reply:
sounds good, just make sure that the same logs are not added twice ;-)

Reply:
I've checked that now :)